props/prop_base.h

```diff
@@ -435,12 +432,57 @@ class PropBase : CommandParser, Looper, protected SaberBase {
 #else
     SaberBase::SetVariation(0);
 #endif
+  }
 
+  // Select preset (font/style)
+  virtual void SetPreset(int preset_num, bool announce) {
+    TRACE(PROP, "start");
+    bool on = BladeOff();
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+    UnsetCurrentStyles();
+    if (announce) {
+      AnnouncePreset();
+    }
+    SetBladeLength();
+    SetBladeStyle();
     if (on) On();
     if (announce) SaberBase::DoNewFont();
     TRACE(PROP, "end");
   }
 
+  // Update Blade Style (no On/Off for use in Edit Mode)
+  void UpdateStyle(int preset_num) {
+    TRACE(PROP, "start");
+    SaveColorChangeIfNeeded();
+    // First free all styles, then allocate new ones to avoid memory
+    // fragmentation.
+    UnsetCurrentStyles();
+    SetBladeLength();
+    SetBladeStyle();
+    TRACE(PROP, "end");
+  }
+
+  // Update Font / Save Style in Edit Mode, skips Preon effect (except for Preon Editing previews) using FastOn
+  void UpdateFont(int preset_num, bool preon) {
```

Comment:
I'm not sure why you would every want the "preon" to play if you're updating the font?

Reply:
I don't think UpdateFont() should take a preset number as an argument. That's just an artifact from SetPreset()
Same for UpdateStyle.

Reply:
Wouldn't it be enough to just do Off(); On(); in this case?